### PR TITLE
feat(nemotron-omni): add Megatron Nano image GRPO support

### PR DESCRIPTION
**Pull request by aroshanghias-nvd**
# What does this PR do ?

Add Megatron-based Nemotron Nano Omni image-GRPO support using a dedicated, model-owned multimodal packing path.

> **Cross-repository dependency:** This draft depends on [NVIDIA-NeMo/Megatron-Bridge#4885](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4885). The submodule is temporarily pinned to that PR's head (`ccf4b09e`). After the MBridge PR merges, this PR must be updated to the resulting commit on MBridge `main` before it is marked ready for review.

The integration follows the same high-level boundary used by other maintained VLMs: NeMo-RL supplies the full processor-expanded sequence and multimodal inputs, while the MBridge `NemotronOmniModel` inserts media embeddings and owns sequence packing and context-parallel sharding. It does not use the historical Nemotron Omni `LLaVAModel` collapse/expand path.

The PR includes:

- capability-based dispatch for model-owned sequence and MTP-mask packing;
- a clear error for legacy LLaVA-layout Nemotron Omni checkpoints;
- dynamic-resolution and historical tiled-image metadata handling;
- NeMo-side generation compatibility for raw policy logprobs, context-length capping, RADIO LayerScale initialization, and cached `bad_words` tokenization;
- maintained single-node Clevr and four-node MMPR Nano image-GRPO recipes.

No external vLLM source code or vLLM dependency pin is changed. The vLLM compatibility behavior is implemented in NeMo-RL's existing integration and runtime-patch layer.

# Issues

No public issue.

# Usage

The maintained recipes added by this PR are:

- `examples/configs/recipes/vlm/vlm_grpo-nemotron-omni-30ba3b-clevr-1n8g-megatron-tp8ep8.v1.yaml`
- `examples/configs/recipes/vlm/vlm_grpo-nemotron-omni-30ba3b-mmpr-4n8g-megatron-tp8ep16.v1.yaml`

# Validation

- Matched production-shaped Omni-versus-main Nano image-GRPO runs exercised generation, rewards, policy logprobs, training, repeated vLLM refit, and checkpointing. Results are available in the [Nemotron Omni main-migration W&B project](https://wandb.ai/joc/nemotron-omni-main-migration).
- Focused default-lane changed tests: `28 passed`.
- Focused vLLM backend tests: `4 passed`.
- The changed MCore data-path tests passed before the existing local test environment reached its missing `megatron.energon` dependency. The corresponding exact-head MBridge change is separately covered by `89` focused unit tests and a ten-iteration functional recipe in the dependency PR. Protected NeMo-RL CI remains required before this draft is marked ready.
- Pre-commit formatting, Ruff, Markdown filename, and recipe-minimization hooks: passed. Pyrefly is currently blocked by three pre-existing W&B type-stub errors in files not changed by this PR.
- `git diff --check`: passed.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [x] Did you add or update any necessary documentation? Updated `docs/guides/nemotron-3-nano-omni.md` with the Megatron architecture, supported scope, checkpoint compatibility, recipes, and usage.

# Additional Information

This PR is intentionally a draft. It must remain draft until the MBridge dependency merges, the submodule is repinned to MBridge `main`, and protected CI passes.

### CODE REVIEW

File: nemo_rl/data/multimodal_utils.py [hunk 1]
```diff
@@ -109,8 +113,28 @@ def as_tensor(
...
+            and len({t.shape[1] for t in non_none_tensors}) == 1
+        ):
+            max_height = max(t.shape[2] for t in non_none_tensors)
+            max_width = max(t.shape[3] for t in non_none_tensors)
+            non_none_tensors = [
+                F.pad(t, (0, max_width - t.shape[3], 0, max_height - t.shape[2]))
+                for t in non_none_tensors
+            ]
+
+        return torch.cat(non_none_tensors, dim=self.dim_to_pack).to(device)
```
Comment (rohitrango) [inline nemo_rl/data/multimodal_utils.py:160]: seems a little specific for 2d images. will it generalize to video frames, audio embeddings, etc.? `as_tensor` is called for all types of tensors imo

Comment (aroshanghias-nvd) [reply nemo_rl/data/multimodal_utils.py:160]: Good point. The original implementation was opt-in, but the padding logic itself assumed 4-D NCHW tensors. I generalized `pad_to_max_shape` to pad every non-packing dimension for same-rank tensors, so the mechanism now supports image, video, and audio/feature shapes. I added 3-D audio-like and 5-D video-like tests alongside the existing dynamic-image coverage in df0db31e. The current caller remains explicitly scoped to `pixel_values`; other modalities can opt in when needed.

**Comment by aroshanghias-nvd** [comment]
Superseded by #3290, which uses an upstream-hosted branch in NVIDIA-NeMo/RL. The replacement starts from the identical head commit (`df0db31e`) and includes the resolved multimodal-padding review fix. Please continue review and CI on #3290.
